core/src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -740,6 +763,17 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     loadConfig[ConfigWithDouble](conf).failure.exception.getMessage shouldEqual "Cannot read a Double from an empty string."
   }
 
+  "Converting from a wrong system properties list" should "complain about having the wrong sytem properties list syntax" in {
+    System.setProperty("pure.conf.intSet.0", "1")
+    System.setProperty("pure.conf.intSet.a", "2")
+    val expected = Set(1, 2)
+
+    assert(loadConfig[Set[Int]]("pure.conf.intSet").failure.exception.getMessage.contains("Cannot parse this object as list! Expecting syntax like"))
+
+    System.clearProperty("pure.conf.intSet.0")
+    System.clearProperty("pure.conf.intSet.a")
+  }
+
```

Comment:
Please add a test which defines the behavior if an index is skipped. For example, what happens when the configuration looks like:

```
val cf = ConfigFactory.parseString("""
  pure.conf: {
    // Note there's no 0 index
    intSet.1: 2  
    intSet.2: 3
  }""")
```